### PR TITLE
made dither layer unique to prevent hard edges

### LINKED ISSUE #4280: Dither Layer for terrains should be unique over all 4 worlds
The Terrains "Summer_swamp" and "wasteland_hard_ground_2" have the same dither layer resulting in sharp edges of triangles 

See Forum Thread https://www.widelands.org/forum/topic/4964/

Other terrains to be checked

### PR DESCRIPTION
**Pull request by hessenfarmer**
fixes #4280

### CODE REVIEW

**Comment by Noordfrees** [comment]
LGTM, but IMHO we should code a check in C++ to ensure that no two terrains have the same layer as this would regress easily

**Comment by hessenfarmer** [comment]
> 
> 
> LGTM, but IMHO we should code a check in C++ to ensure that no two terrains have the same layer as this would regress easily

+1 for this however I am not good in c++ coding so I would appreciate if somebody else with more experience could do this.

**Comment by Noordfrees** [comment]
Okay I can code such a check, but it might be better to wait with this until #4240 is in to avoid merge conflicts

**Comment by Noordfrees** [comment]
Easier than I thought, implemented this now :)

File: src/logic/map_objects/world/terrain_description.cc [hunk 1]
```diff
@@ -115,7 +115,8 @@ TerrainDescription::TerrainDescription(const LuaTable& table, Widelands::World&
 	for (DescriptionIndex di = world.get_nr_terrains(); di; --di) {
 		const TerrainDescription& t = world.terrain_descr(di - 1);
 		if (t.dither_layer_ == dither_layer_) {
-			throw GameDataError("Terrain %s has the same dither layer %i as %s", name_.c_str(), dither_layer_, t.name_.c_str());
+			throw GameDataError("Terrain %s has the same dither layer %i as %s", name_.c_str(),
+			                    dither_layer_, t.name_.c_str());
```
Comment (SuperSandro2000) [inline src/logic/map_objects/world/terrain_description.cc:119]: mix of tabs and spaces

Comment (hessenfarmer) [reply src/logic/map_objects/world/terrain_description.cc:119]: the spaces are inserted by our fix_formatting script and are ok from my side. 
So for me code looks good.

**Comment by hessenfarmer** [comment]
@Noordfrees how shall we proceed from my side I can approve your changes. But I shouldn't approve mine

**Comment by Noordfrees** [comment]
Then we have hereby approved each others' changes :) So I'll merge now

**Pull request merged**